movies/src/components/header/Header.js

```diff
@@ -1,23 +1,77 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Logo from '../common/Logo';
 import Search from '../search/Search';
 import AddButton from '../addButton/AddButton';
+import MovieCard from '../movieCard/MovieCard';
+import SearchButton from '../searchButton/SearchButton';
 
-export default function Header() {
+const setMovieGenre = (movie) => movie.genre.join(', ');
+
+export default function Header(props) {
+  const { isMovieSelected, selectedMovie, goHomePage } = props;
   return (
     <header className="header">
       <div className="header__content">
         <div className="header-line">
           <Logo />
-          <AddButton />
+
+          {
+            !isMovieSelected ?
+              <AddButton />
```

Comment:
in `AddButton` `hidePopup` and `showPopup` cound be wrapped with useCallback

Reply:
However, this is not mandatory https://dmitripavlutin.com/dont-overuse-react-usecallback/

Reply:
Fixed